core/conn/jdbc_type2/native/SrvrCommon.cpp

```diff
@@ -1634,6 +1635,53 @@ short do_ExecSMD(
                         tableParam[0], inputParam[0], inputParam[1],
                         inputParam[2], inputParam[3]);
                     break;
+	case SQL_API_SQLSTATISTICS:
+		    if ((!checkIfWildCard(catalogNm, catalogNmNoEsc) || !checkIfWildCard(schemaNm, schemaNmNoEsc)
+		        || !checkIfWildCard(tableNm, tableNmNoEsc)) && !metadataId)
+		    {
+		        executeException->exception_nr = odbc_SQLSvc_GetSQLCatalogs_ParamError_exn_;
+		        executeException->u.ParamError.ParamDesc = SQLSVC_EXCEPTION_WILDCARD_NOT_SUPPORTED;
+		        FUNCTION_RETURN_NUMERIC(EXECUTE_EXCEPTION,("EXECUTE_EXCEPTION"));
+		    }
+
+		    if (strcmp(catalogNm,"") == 0)
+		        strcpy(tableName1,SEABASE_MD_CATALOG);
+		    else
+		        strcpy(tableName1, catalogNm);
+		    tableParam[0] = tableName1;
+		    convertWildcard(metadataId, TRUE, schemaNm, expSchemaNm);
+		    convertWildcardNoEsc(metadataId, TRUE, schemaNm, schemaNmNoEsc);
+		    convertWildcard(metadataId, TRUE, tableNm, expTableNm);
+		    convertWildcardNoEsc(metadataId, TRUE, tableNm, tableNmNoEsc);
+		    if( inputParam[5] != NULL )
+		    {
+		        sprintf(cunique, " and index.IS_UNIQUE=1");
+		    }
+		    inputParam[0] = schemaNmNoEsc;
+		    inputParam[1] = expSchemaNm;
+		    inputParam[2] = tableNmNoEsc;
+		    inputParam[3] = expTableNm;
+		    inputParam[4] = NULL;
+		    snprintf((char *)sqlString->dataValue._buffer, totalSize,
+		    "select "
+		    "obj.CATALOG_NAME TABLE_CAT, obj.SCHEMA_NAME TABLE_SCHEM, obj.OBJECT_NAME TABLE_NAME,"
+		    "cast((case when index.IS_UNIQUE=0 then 1 else 0 end) as smallint) NON_UNIQUE, "
+		    "cast(NULL as varchar(32)) INDEX_QUALIFIER, idxobj.OBJECT_NAME INDEX_NAME, "
+		    "cast(-2 as smallint) TYPE, cols.COLUMN_NUMBER ORDINAL_POSITION, "
+		    "substr(cols.COLUMN_NAME,0,CHAR_LENGTH(cols.COLUMN_NAME)) COLUMN_NAME, "
+		    "cast((case when keys.ORDERING=0 then 'A' else 'D' end) as varchar(4)) ASC_OR_DESC,"
+		    "cast(-2 as int) CARDINALITY, cast(-2 as int) PAGES, cast(NULL as varchar(10)) FILTER_CONDITION "
```

Comment:
Says in standard API doc: 
CARDINALITY int => When TYPE is tableIndexStatistic, then this is the number of rows in the table; otherwise, it is the number of unique values in the index. 
PAGES int => When TYPE is tableIndexStatisic then this is the number of pages used for the table, otherwise it is the number of pages used for the current index. 
TYPE short => index type: ◦ tableIndexStatistic - this identifies table statistics that are returned in conjuction with a table's index descriptions 
◦ tableIndexClustered - this is a clustered index 
◦ tableIndexHashed - this is a hashed index 
◦ tableIndexOther - this is some other style of index